packages/language-server/src/server.ts

```diff
@@ -0,0 +1,156 @@
+import {
+    IConnection,
+    TextDocuments,
+    DiagnosticSeverity,
+    Diagnostic,
+    createConnection,
+    IPCMessageReader,
+    IPCMessageWriter
+} from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import * as MessageHandler from './MessageHandler'
+import { fullDocumentRange } from './provider'
+import * as util from './util'
+import lint from './lint'
+import fs from 'fs'
+import install from './install'
+
+export interface LSOptions {
+    /**
+     * If you have a connection already that the ls should use, pass it in.
+     * Else the connection will be created from `process`.
+     */
+    connection?: IConnection;
+}
+
+/**
+ * Starts the language server.
+ *
+ * @param options Options to customize behavior
+ */
+export function startServer(options?: LSOptions) {
+    let connection = options?.connection;
+    if (!connection) {
```

Comment:
If you put the "let's get a connection" into a separate function that ensures that a connection is returned, you don't need the `connection!...` later.